client_authentication.go

```diff
@@ -251,6 +262,40 @@ func (f *Fosite) DefaultClientAuthenticationStrategy(ctx context.Context, r *htt
 	return client, nil
 }
 
+// Tries to read client ID from (yet) unverified token, if request came unauthenticated
+func jwtBearerNoClientAuthRequest(r *http.Request, form url.Values, originalErr error) (clientID, clientSecret string, err error) {
+	if grantType := form.Get("grant_type"); grantType != grantTypeJWTBearer {
+		return "", "", originalErr
+	}
+	assertion := form.Get("assertion")
+
+	if assertion == "" {
+		return "", "", originalErr
+	}
+	// do not validate the key (hence noop token func), all this is just to get the issuer as client ID
```

Comment:
Do we check if the assertion is a single JWT?

Reply:
Assertion is retrieved through form.Get("assertion") which gets the first value associated with the given key.

Reply:
I'm referring to RFC mentioning   `The value of the "assertion" parameter MUST contain a single JWT.`

Reply:
What is the concrete risk you see here? If form contains multiple assertions the first one is taken. If assertion parameter has multiple JWTs contained, the subsequent Parsing will fail. Hence I think we have sufficient assurance. However, if you insist I can check whether the form's assertion array has length 1 and if > 1 raise an error.

Reply:
there's no concrete risk, I'm getting familiar with code base here in general and at the same time I went through the RFC you have linked ;-)  I've looked into the entire file and see it's always done using form.Get() without checking if multiple assertions exist.